game-core/src/system/ally.rs

```diff
@@ -13,9 +15,70 @@ pub struct Movement;
 impl<'s> System<'s> for Movement {
     type SystemData = (ReadStorage<'s, Ally>, WriteStorage<'s, Transform>);
 
-    fn run(&mut self, (players, mut transforms): Self::SystemData) {
-        for (_, transform) in (&players, &mut transforms).join() {
-            println!("ally: {:?}", transform);
+    fn run(&mut self, (allies, mut transforms): Self::SystemData) {
+        for (_, transform) in (&allies, &mut transforms).join() {}
+    }
+}
+
+pub struct Grouper;
+
+impl<'s> System<'s> for Grouper {
+    type SystemData = (
+        ReadStorage<'s, Ally>,
+        WriteStorage<'s, Transform>,
+        WriteStorage<'s, Parent>,
+        WriteStorage<'s, Player>,
+        Entities<'s>,
+    );
+
+    fn run(
+        &mut self,
+        (allies, mut transforms, mut parents, mut players, entities): Self::SystemData,
+    ) {
+        let get_relative_position = |mut num_sprites: u32| {
+            num_sprites %= 9;
+
+            let ind = if num_sprites == 4 {
```

Comment:
Is this case related to the player?
Might need some comment about it.

Reply:
yea its hopping past the player spot